packages/contracts/contracts/test/AssetManager.t.sol

```diff
@@ -0,0 +1,608 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+import { L2OutputOracle_Initializer } from "./CommonTest.t.sol";
+import { Constants } from "../libraries/Constants.sol";
+import { Types } from "../libraries/Types.sol";
+import { Predeploys } from "../libraries/Predeploys.sol";
+import { Uint128Math } from "../libraries/Uint128Math.sol";
+import { Colosseum } from "../L1/Colosseum.sol";
+import { L2OutputOracle } from "../L1/L2OutputOracle.sol";
+import { ValidatorPool } from "../L1/ValidatorPool.sol";
+import { AssetManager } from "../L1/AssetManager.sol";
+import { TestERC721 } from "./L1ERC721Bridge.t.sol";
+import { IKGHManager } from "../universal/IKGHManager.sol";
+import { Proxy } from "../universal/Proxy.sol";
+
+contract MockL2OutputOracle is L2OutputOracle {
+    constructor(
+        ValidatorPool _validatorPool,
+        address _colosseum
+    ) L2OutputOracle(ValidatorPool(address(0)), address(_colosseum), 1800, 2, 0, 0, 7 days) {}
+
+    function addOutput(bytes32 _outputRoot, uint256 _l2BlockNumber) external payable {
+        l2Outputs.push(
+            Types.CheckpointOutput({
+                submitter: msg.sender,
+                outputRoot: _outputRoot,
+                timestamp: uint128(block.timestamp),
+                l2BlockNumber: uint128(_l2BlockNumber)
+            })
+        );
+    }
+
+    function replaceOutput(uint256 _outputIndex, bytes32 _outputRoot) external {
+        l2Outputs[_outputIndex] = Types.CheckpointOutput({
+            submitter: msg.sender,
+            outputRoot: _outputRoot,
+            timestamp: uint128(block.timestamp),
+            l2BlockNumber: l2Outputs[_outputIndex].l2BlockNumber
+        });
+    }
+}
```

Comment:
What about importing MockL2OutputOracle from another test?

Reply:
That would be better, I'll change it. Thanks!